### DISCUSSION TITLE
Is the menu option "Run > Start Debugging (F5)" supposed to run the build task (MakePQX.exe compile)?

### DISCUSSION THREAD

**Discussion by paulsinz** [microsoft/vscode-powerquery-sdk#278] [Q&A]
The Start Debugging menu option in Visual Studio Code does not rebuild the .mez file, even if changes have been made to the .pq file. Is this the desired behavior?

In Visual Studio Code the Power Query SDK provides an Explorer sidebar dropdown option of "Evaluate current file" which does rebuild the .pq file. The documentation found [here](https://learn.microsoft.com/en-us/power-query/power-query-sdk-vs-code#evaluate-a-query-and-the-results-panel) seems to indicate that both evaluation approaches are equivalent without mentioning that Start Debugging potentially requires a user to rebuild their connector file.

Potential Confusion for New Users
--------------------------------------
When following the TripPin walkthrough ([here](https://learn.microsoft.com/en-us/power-query/samples/trippin/1-odata/readme)) from section "1. OData" to section "2. REST API", unexpected behavior is encountered if the user only uses the Start Debugging evaluation approach since the .mez file is not rebuilt. Specifically, the output of the TripPin.query.pq file is not as described in the walkthrough.

**Comment by kozmaz87** [comment]
I ran into this as well. It is not being rebuilt no matter how you want to run it. The only way I could trigger a rebuild via the ui is when I use the Run TestConnection function  in the Power Query SDK subtab.
